packages/bot-web-ui/src/stores/summary-card-store.ts

```diff
@@ -204,6 +207,26 @@ export default class SummaryCardStore {
         }
     }
 
+    /**
+     * Sets the bot's running state based on whether the contract is still loading
+     */
+    setIsBotRunning() {
+        if (!this.is_contract_loading) {
+            this.is_bot_running = false;
+            return;
+        }
+
+        const onTimeout = () => {
+            if (this.is_contract_loading) {
+                this.is_bot_running = true;
+                this.root_store.run_panel.setContractStage(contract_stages.RUNNING);
+            }
+        };
+
+        const timeout = setTimeout(onTimeout, 5000);
+        return () => clearTimeout(timeout);
+    }
```

Comment:
@maryia-matskevich-deriv what is the use case of this function? Is there any particular reason to add setTimeout?

Reply:
@shafin-deriv we need it to follow to requirements:
"`The panel here will show this status - "Bot is starting", when user clicks on Run and the Summary tab will show a skeleton loader for a maximum of 5 seconds.`" Only after **5 seconds** we show a loader with the appropriate message.